cmd/login/wc.go

```diff
@@ -90,52 +61,138 @@ func (r *runner) createClusterClientCert(ctx context.Context, client k8sclient.I
+	}
+	return c, nil
+}
+
+func (r *runner) getCertOperatorVersion(c *cluster.Cluster, provider string, services serviceSet, ctx context.Context) (string, error) {
 	releaseVersion, err := getClusterReleaseVersion(c, provider, r.flag.WCInsecureNamespace)
 	if err != nil {
-		return microerror.Mask(err)
+		return "", microerror.Mask(err)
 	}
+	certOperatorVersion, err := getCertOperatorVersion(ctx, services.releaseService, releaseVersion)
+	// If the release does not contain cert-operator anymore (e.g. in CAPI versions) we return an empty string
+	// In this case we try to create the client certificate using the MC PKI
+	if IsMissingComponent(err) {
+		return "", nil
+	} else if err != nil {
+		return "", microerror.Mask(err)
+	}
+	return certOperatorVersion, nil
+}
+
+func (r *runner) handleWCLogin(ctx context.Context) error {
+	// At the moment, the only available login option for WC is client cert
+	return r.handleWCClientCert(ctx)
+}
 
-	certOperatorVersion, err := getCertOperatorVersion(ctx, releaseService, releaseVersion)
+func (r *runner) handleWCClientCert(ctx context.Context) error {
+	var err error
+
+	config := commonconfig.New(r.flag.config)
+
+	provider, err := config.GetProvider()
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	clusterBasePath, err := getClusterBasePath(r.k8sConfigAccess)
+	var client k8sclient.Interface
+	{
+		client, err = config.GetClient(r.logger)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+	// At the moment, the only available login option for WC is client cert
+	contextName, contextExists, err := r.createClusterClientCert(ctx, client, provider)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
+	if r.loginOptions.selfContainedWC {
+		fmt.Fprintf(r.stdout, "A new kubectl context has been created named '%s' and stored in '%s'. You can select this context like this:\n\n", contextName, r.flag.SelfContained)
+		fmt.Fprintf(r.stdout, "  kubectl cluster-info --kubeconfig %s \n", r.flag.SelfContained)
+	} else if !r.loginOptions.switchToWCcontext {
+		fmt.Fprintf(r.stdout, "A new kubectl context has been created named '%s'. To switch back to this context later, use this command:\n\n", contextName)
+		fmt.Fprintf(r.stdout, "  kubectl config use-context %s\n", contextName)
+	} else if contextExists {
+		fmt.Fprintf(r.stdout, "Switched to context '%s'.\n\n", contextName)
+		fmt.Fprintf(r.stdout, "To switch back to this context later, use this command:\n\n")
+		fmt.Fprintf(r.stdout, "  kubectl config use-context %s\n", contextName)
+	} else {
+		fmt.Fprintf(r.stdout, "A new kubectl context has been created named '%s' and selected. To switch back to this context later, use this command:\n\n", contextName)
+		fmt.Fprintf(r.stdout, "  kubectl config use-context %s\n", contextName)
+	}
+
+	return nil
+}
+
+func (r *runner) createClusterClientCert(ctx context.Context, client k8sclient.Interface, provider string) (string, bool, error) {
+	var err error
+
+	err = validateProvider(provider)
```

Comment:
```suggestion
	err := validateProvider(provider)
```
No need to declare err at the top of each function especially if you're setting it in the next lines.